### PR TITLE
mariadb-connector-c: fix PLUGINDIR

### PR DESCRIPTION
**Pull request by dvetutnev**
Need relative path
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@dvetutnev you have to ask EAP access here https://github.com/conan-io/conan-center-index/issues/4

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
could you please explain what this solves ?

File: recipes/mariadb-connector-c/all/test_package/test_package.c [hunk 1]
```diff
@@ -1,11 +1,11 @@
 #include <mysql.h>
 #include <stdio.h>
 
-// See #4530
-#include <mysqld_error.h>
 
 int main(int argc, char **argv)
 {
   printf("MySQL client version: %s\n", mysql_get_client_info());
+  printf("MARIADB_PLUGINDIR: %s", MARIADB_PLUGINDIR);
```
Comment (ericLemanissier) [inline recipes/mariadb-connector-c/all/test_package/test_package.c:8]:
```suggestion
  printf("MARIADB_PLUGINDIR: %s\n", MARIADB_PLUGINDIR);
```

**Review by ericLemanissier — APPROVED** [review]
ok, IIUC, setting cmake variable `INSTALL_PLUGINDIR` to an absolute path hardcodes the macro `MARIADB_PLUGINDIR`  to the absolute package path, which is not correct when you download a pre-built binary.

File: recipes/mariadb-connector-c/all/test_package/test_package.c [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #include <mysql.h>
 #include <stdio.h>
 
-// See #4530
-#include <mysqld_error.h>
```
Comment (prince-chrismc) [inline recipes/mariadb-connector-c/all/test_package/test_package.c:5]: Why is this removed? it's a test for bug some consumers faced 🤔

Comment (dvetutnev) [reply recipes/mariadb-connector-c/all/test_package/test_package.c]: Restore

**Review by prince-chrismc — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]
